### PR TITLE
test: build docker image

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* Push the Docker images generated to the observability-ci namespace in our Docker registry.
* Limit the changes that trigger a full build on the main pipeline.

## Why is it important?

After this PR we will have fresh snapshots every time we run the packaging job.
Also, changes in the packaging job will no trigger a full build of the main pipeline.

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -124,11 +124,49 @@ pipeline {
...
+  def oldName = "${DOCKER_REGISTRY}/beats/${name}:${libbetaVer}"
+  def newName = "${DOCKER_REGISTRY}/observability-ci/${name}:${libbetaVer}"
+  def commitName = "${DOCKER_REGISTRY}/observability-ci/${name}:${env.GIT_BASE_COMMIT}"
+  dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
+  sh(label:'Change tag and push', script: """
+    docker tag ${oldName} ${newName}
+    docker push ${newName}
+    docker tag ${oldName} ${commitName}
+    docker push ${commitName}
+  """)
```
Comment (v1v) [inline .ci/packaging.groovy:169]: what about retry/sleep to avoid any issues with the internal docker registry?

Comment (kuisathaverat) [inline .ci/packaging.groovy:169]:
```suggestion
  retry(3){
    sh(label:'Change tag and push', script: """
      docker tag ${oldName} ${newName}
      docker push ${newName}
      docker tag ${oldName} ${commitName}
      docker push ${commitName}
    """)
  }
```

Comment (kuisathaverat) [reply .ci/packaging.groovy:169]: yep, I do not like the retries but yep we can add it

**Comment by ph** [comment]
@kuisathaverat How do I test that PR?

**Comment by kuisathaverat** [comment]
the comment `/packaging` trigger the job related to this PR, we are pending of a change in the job to notify a GitHub check, so right now the only way to see the result it is to go to the Job PR https://beats-ci.elastic.co/job/Beats/job/packaging/view/change-requests/job/PR-17905/

**Comment by kuisathaverat** [comment]
@andrewkroh any thoughts?

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -124,11 +124,51 @@ pipeline {
...
+      tagAndPush('heartbeat-oss')
+    } else if ("${env.BEATS_FOLDER}" == "journalbeat"){
+      tagAndPush('journalbeat')
+      tagAndPush('journalbeat-oss')
+    } else if ("${env.BEATS_FOLDER}" == "metricbeat"){
+      tagAndPush('metricbeat-oss')
+    } else if ("${env.BEATS_FOLDER}" == "packetbeat"){
+      tagAndPush('packetbeat')
+      tagAndPush('packetbeat-oss')
+    } else if ("${env.BEATS_FOLDER}" == "x-pack/auditbeat"){
```
Comment (mdelapenya) [inline .ci/packaging.groovy:143]: A possible simplification for the if/else branches (if and only if groov-ish supports it) 🤔 
```groovy
if ("${env.BEATS_FOLDER}".startsWith("x-pack/"){
      def theBeat = "${env.BEATS_FOLDER}".replaceAll("x-pack/", "")
      tagAndPush(theBeat)
}
```

Comment (kuisathaverat) [reply .ci/packaging.groovy:143]: I did not see a clear rule some beats build OSS and x-pack, others only OSS, some only x-pack

**Pull request merged**